x-pack/solutions/security/plugins/security_solution/public/management/components/policy_selector/hooks/use_fetch_policy_data.ts

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { UseQueryResult } from '@tanstack/react-query';
+import type { GetPackagePoliciesResponse } from '@kbn/fleet-plugin/common';
+import type { IHttpFetchError } from '@kbn/core-http-browser';
+import { useMemo } from 'react';
+import { chunk } from 'lodash';
+import type { PolicyData } from '../../../../../common/endpoint/types';
+import type { PolicySelectorProps } from '..';
+import { useFetchIntegrationPolicyList } from '../../../hooks/policy/use_fetch_integration_policy_list';
+import { useBulkFetchFleetIntegrationPolicies } from '../../../hooks/policy/use_bulk_fetch_fleet_integration_policies';
+
+/**
+ * Hook for use in the PolicySelector component. It normalizes the retrieval of data between
+ * just fetching the regular list of policies -OR- bulk retrieving a list of policy by IDs.
+ *
+ * The primary goal of this hook is to efficiently manage the display of selected policies, which
+ * uses the Bulk Get Package Policies API from fleet which does not have support for pagination.
+ */
+export const useFetchPolicyData = (
+  queryOptions: PolicySelectorProps['queryOptions'] & { page: number },
+  selectedPolicyIds: PolicySelectorProps['selectedPolicyIds'],
+  mode: 'full-list' | 'selected-list'
+): Pick<
+  UseQueryResult<GetPackagePoliciesResponse, IHttpFetchError>,
+  'data' | 'isFetching' | 'isLoading' | 'error'
+> => {
+  const selectedPoliciesPagination: string[][] = useMemo(() => {
+    if (mode === 'selected-list') {
+      return chunk(selectedPolicyIds, queryOptions?.perPage ?? 20);
```

Comment:
nit: the `20` is used two times, it could be a default value for the optional `perPage` attribute

Reply:
Not understanding the comment here. Can you clarify how its being used twice?

I did not destruct the `queryOptions` in the function call arguments because I'm just passing that as is to the Fetch policy list.

Reply:
yeah, it was a bit unclear, sorry 😅 so it's duplicated in lines 35 and 65